hack/reattach-pv/README.md

```diff
@@ -0,0 +1,54 @@
+# Reattach-PV
+
+This tool can be used to recreate an Elasticsearch cluster by reusing existing PersistentVolumes, that belonged to that cluster before it was deleted.
+
+## Expectations
+
+This tool can only be used when the following conditions are met:
+
+* The Elasticsearch resource to re-create does not exist in Kubernetes.
+* All PersistentVolumeClaims of the previous cluster do not exist anymore.
+* All PersistentVolumes of the previous cluster still exist. Their status report they are `Released`.
+* The Elasticsearch resource to re-create has the exact same specs as the deleted one. Same cluster name, same node sets, same count, etc.
+* The current default kubectl context targets the desired Kubernetes cluster.
+
+## Usage
+
+```
+Recreate an Elasticsearch cluster by reattaching existing released PersistentVolumes
+
+Usage:
+  reattach-pv [flags]
+
+Flags:
+      --dry-run                         do not apply any Kubernetes resource change
+      --elasticsearch-manifest string   path pointing to the Elasticsearch yaml manifest
+  -h, --help                            help for reattach-pv
+      --pv-backup-path string           path to the file where a backup of existing PersistentVolumes will be stored before update, set empty to disable (default "pv_backup_{timestamp}.json")
```

Comment:
To me, the wording here implies that the actual contents of the volumes are copied to make a backup of the data. However, the tool only dumps the PV definition in JSON to a file. Is there any benefit to having the definition backed up?  

In my mind, the risk here is that the restoration operation ends up corrupting the data already stored in the volumes. The only way to mitigate that is to clone the disks (gcloud supports snapshots, not sure about other providers) before the tool is run.

Reply:
You mark a point here. I had in mind the PVs JSON backup could be useful to investigate why something went wrong with the new cluster, by investigating the patched PVs vs. the ones we had before the script ran. But as you point out data is probably already corrupted at this point.

Let's maybe remove the backup feature altogether?

Reply:
Maybe add a line to the README to say that it is the responsibility of the users to backup the underlying disks before attempting to run the tool?